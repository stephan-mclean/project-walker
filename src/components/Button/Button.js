import React from "react";

const Button = props => <button {...props} />;

Button.defaultProps = {
  type: "button"
};

export default Button;
