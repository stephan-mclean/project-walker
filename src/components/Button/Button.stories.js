import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";

import Button from "./Button";

storiesOf("Components", module).add("Button", () => {
  return (
    <Fragment>
      <Button className="btn">Default</Button>
      <Button className="btn btn--primary">Primary</Button>
      <Button className="btn btn--danger">Danger</Button>
      <Button className="btn btn--primary" disabled>
        Disabled
      </Button>
      <Button className="btn btn--outline">Outline</Button>
      <Button className="btn btn--inverted">Inverted</Button>
      <Button className="btn btn--link">Link</Button>
    </Fragment>
  );
});
