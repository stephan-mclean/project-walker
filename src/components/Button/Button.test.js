import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("Testing Button component", () => {
  it("Should match the snapshot", () => {
    const button = shallow(
      <Button className="btn btn--primary">Button</Button>
    );

    expect(button).toMatchSnapshot();
  });

  it("Should render the correct text", () => {
    const button = shallow(
      <Button className="btn btn--primary">My Button</Button>
    );

    expect(button.text()).toEqual("My Button");
  });

  it("Should respond to the users click", () => {
    const onClick = jest.fn();
    const button = shallow(
      <Button className="btn btn--primary" onClick={onClick}>
        My Button
      </Button>
    );

    button.simulate("click");

    expect(onClick).toHaveBeenCalled();
  });
});
