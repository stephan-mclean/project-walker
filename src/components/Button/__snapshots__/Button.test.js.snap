// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Should match the snapshot 1`] = `
<button
  className="btn btn--primary"
  type="button"
>
  Button
</button>
`;
